File: docs/code-quality/codesnippet/CSharp/ca2208-instantiate-argument-exceptions-correctly_2.cs
namespace Samples2
{    
    public class Book    
    {        
        private readonly string _Title;
 
        public Book(string title)        
        {            
            if (title == null)                
                throw new ArgumentNullException("title", "title cannot be a null reference (Nothing in Visual Basic)");
 
            _Title = title;        }
 
        public string Title        
        {            
            get { return _Title; }        
        }
    }
}

File: docs/code-quality/codesnippet/CPP/ca2208-instantiate-argument-exceptions-correctly_2.cpp
using namespace System;
 
namespace Samples2 
{    
	public ref class Book    
	{
     private: initonly String^ _Title;
 
    public:
        Book(String^ title)        
		{            
			if (title == nullptr)                
				throw gcnew ArgumentNullException(("title", "title cannot be a null reference (Nothing in Visual Basic)"));
 
            _Title = title;        
		}
 
        property String^ Title        
		{            
			String^ get()            
			{                
				return _Title;            
			}        
		}    
	};
}